frontend/src/shared/helpers/routing/reverse.ts

```diff
@@ -1,5 +1,24 @@
 import { Path, paths } from '@shared/consts/routing';
+import { UserActivationParams } from '@users/types/activation';
+import { isString, replace } from 'lodash';
 
-export const reverse = (path: Path): string => {
-  return paths[path];
+interface DetailedPath<TPath extends string, TParams extends Record<string, any>> {
+  path: TPath;
+  params: TParams;
+}
+
+type RegularPaths = Exclude<Path, 'users.activation'>;
+
+type ReversiblePath = RegularPaths | DetailedPath<'users.activation', UserActivationParams>;
+
+export const reverse = (path: ReversiblePath): string => {
+  if (isString(path)) return paths[path as Path];
```

Comment:
Hmm, moze to jest spoko, nie wiem, ale odkad kodze nigdy nie widzialem takiego zapisu. Zawsze wygladalo to tak

```
if (costam) {
  ...
} else {
   ...
}
```

Poza tym nie potrzebujesz tego `else`, skoro w ifie wychodzisz z funkcji (`return`)